server/src/main/java/org/elasticsearch/plugins/PluginsService.java

```diff
@@ -293,6 +294,34 @@ static void loadExtensions(Collection<LoadedPlugin> plugins) {
         }
     }
 
+    // package private for testing
+    <T> Iterator<T> providersIterator(Class<T> service, ClassLoader classLoader) {
+        return ServiceLoader.load(service, classLoader).iterator();
+    }
+
+    /**
+     * SPI convenience method that uses the {@link ServiceLoader} JDK class to load various SPI providers
+     * from plugins/modules.
+     *
+     * <p>
+     * For example:
+     *
+     * var pluginHandlers = pluginsService.loadServiceProviders(OperatorHandlerProvider.class);
+     * </p>
```

Comment:
nit: I *think* the closing p tag is normally omitted in javadoc (or must be?)

Reply:
I thought it was allowed, I see we use it in few places, but it's stripped now and I'll remove it from other PR.